src/LogCleaner.cc

```diff
@@ -789,13 +789,16 @@ LogCleaner::Balancer::compactionFailed()
 LogCleaner::Balancer::CleaningTask
 LogCleaner::Balancer::requestTask(CleanerThreadState* thread)
 {
-    if (isDiskCleaningNeeded(thread))
+    // TODO: this is a hack replacement for the more comprehensive memory
+    // management logic found in the cleaner.  On a very tight timeline for
+    // the 0.7.3 release so we are just hacking to stop a really bad memory
+    // leak.
+    static uint64_t count = 0;
+    ++count;
+    if (count % 200 == 0) {
         return CLEAN_DISK;
-
-    if (!cleaner->disableInMemoryCleaning && isMemoryLow(thread))
-        return COMPACT_MEMORY;
-
-    return SLEEP;
+    }
+    return count % 25 == 0 ? COMPACT_MEMORY : SLEEP;
```

Comment:
How often is requestTask() called?  Not worried about wraparound of a 64-bit counter, just wondering what that translates to for kicking compaction or cleaning.

Reply:
I have same question, thanks.

Reply:
That is a fair point.  I thought about it but, assuming we increment 5 billion times per second then:

```python3
 >>> 2**64 / (5*10**9) / (24 * 60 * 60 * 365.25)
116.90840921812529
```

it takes 116.9 years for us to wrap our counter.  Likely not worth messing with when we plan to rip out the change as soon as possible.